src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1169,241 @@ const isMobileFieldUpdateAllowed = (
+    new TwilioCredentialsData(twilioCredentials)
+
+  const body: PutSecretValueRequest = {
+    SecretId: msgSrvcName,
+    SecretString: twilioCredentialsData.toString(),
+  }
+
+  return ResultAsync.fromPromise(
+    secretsManager.getSecretValue({ SecretId: msgSrvcName }).promise(),
+    (error) => {
+      logger.error({
+        message: 'Twilio Credentials do not exist in Secrets Manager',
+        meta: {
+          action: 'updateTwilioCredentials',
+          msgSrvcName,
+        },
+        error,
+      })
+
+      return new SecretsManagerNotFoundError(
+        'Twilio Credentials do not exist in Secrets Manager',
+      )
+    },
+  )
+    .andThen(() => {
+      logger.info({
+        message: 'Twilio Credentials has been found in Secrets Manager',
+        meta: {
+          action: 'updateTwilioCredentials',
+          msgSrvcName,
+        },
+      })
+
+      return ResultAsync.fromPromise(
+        secretsManager.putSecretValue(body).promise(),
+        (error) => {
+          logger.error({
+            message: 'Error occurred when updating Twilio in Secret Manager!',
+            meta: {
+              action: 'updateTwilioCredentials',
+              body,
+            },
+            error,
+          })
+
+          return new SecretsManagerError(
+            'Error occurred when updating Twilio in Secret Manager!',
+          )
+        },
+      )
+    })
+    .andThen(() => {
+      return ResultAsync.fromPromise(
+        Promise.resolve(twilioClientCache.del(msgSrvcName)),
+        // Currently, a call to get twilio credentials will cache the credentials in the twilioCache for ~10s
+        // If a call to retrieve twilio credentials occurs before 10s passes, it will be a cache hit, retrieving
+        // the wrong credentials. Hence we need to clear the cache entry
+
+        (error) => {
+          logger.error({
+            message: 'Error occurred when clearing cache in Secret Manager!',
+            meta: {
+              action: 'updateTwilioCredentials',
+              body,
+            },
+            error,
+          })
+
+          return new TwilioCacheError(
+            'Error occurred when clearing cache entry!',
+          )
+        },
+      )
+    })
+}
+
+export const deleteTwilioCredentials = (
+  formId: string,
+  msgSrvcName: string,
+): ResultAsync<unknown, SecretsManagerError | TwilioCacheError> => {
+  const body: DeleteSecretRequest = {
+    SecretId: msgSrvcName,
+    ForceDeleteWithoutRecovery: true,
+    // Need ForceDeleteWithoutRecovery boolean flag for it to be deleted immediately
+    // If not, the key-value pair will remain in SecretsManager for another 14 days before
+    // being dleted: https://docs.aws.amazon.com/secretsmanager/latest/userguide/manage_delete-secret.html
+  }
+
+  return ResultAsync.fromPromise(
+    secretsManager
+      .getSecretValue({
+        SecretId: msgSrvcName,
+      })
+      .promise(),
+    () => {
```

Comment:
is there a specific error for non-existance? We should ignore only that error and return everything else as a real error

Reply:
Currently , the exception we can check for is `ResourceNotFoundException`, but the AWS SDK does not provide its own errors (have to define our own error types).